examples/ui-demo/src/utils/hooks/useDebounceEffect.ts

```diff
@@ -0,0 +1,22 @@
+/**
```

Comment:
Review note:

This was added to deal with the async nature of the Avatar gradient color creation process. Changing the primary color picker could fire multiple times as the user selects a color and could bog down the event loop leading to a poor user experience.